File: elixir/sum_of_multiples.ex
defmodule SumOfMultiples do
  @doc """
  Adds up all numbers from 1 to a given end number that are multiples of the factors provided.
  """
  @spec to(non_neg_integer, [non_neg_integer]) :: non_neg_integer
  def to(limit, factors) do
    factors
    |> Stream.filter(&(&1 != 0))
    |> Enum.reduce(%MapSet{}, fn n, acc ->
      n..(limit - 1)//n
      |> Enum.reduce(acc, fn each, acc ->
        MapSet.put(acc, each)
      end)
    end)
    |> Enum.sum()
  end
end

File: dart/sum_of_multiples.dart
class SumOfMultiples {
  int sum(List<int> arr, int limit) {
    Set<int> st = Set();
    arr.forEach((n) {
      if (n == 0) {
        return;
      }
      for (int i = n; i < limit; i += n) {
        st.add(i);
      }
    });

    if (st.isEmpty) {
      return 0;
    }
    return st.reduce((a, b) => a + b);
  }
}
